### ISSUE TITLE
Issue with Deploy Kubeflow Cluster

### ISSUE THREAD

**Issue by akhilvinvent** [kubeflow/gcp-blueprints#330]
Hi,

I am facing issue with "Deploy Kubeflow Cluster" steps mentioned at below link.

Below upstream directory was not created after running "bash ./pull-upstream.sh" mentioned in document.

gcp-blueprints/kubeflow/apps/pipelines/upstream

https://www.kubeflow.org/docs/distributions/gke/deploy/deploy-cli/

Error: ./gcp-blueprints/kubeflow/apps/pipelines/upstream/third-party/argo/upstream/manifests/namespace-install/overlays/argo-server-deployment.yaml: wrong Node Kind for  expected: MappingNode was SequenceNode: value: {- op: add
  path: /spec/template/spec/containers/0/args/-
  value: --namespaced}

**Comment by edi-bice-by** [comment]
kubeflow apply fails at the install_asm step for me

install_asm: Downloading kpt..
...
install_asm: Downloading ASM..
...
install_asm: Downloading ASM kpt package...
install_asm: [WARNING]: Failed, retrying...(1 of 3)
install_asm: [WARNING]: Failed, retrying...(2 of 3)
install_asm: [WARNING]: Failed, retrying...(3 of 3)
make: *** [install-asm] Error 1

Stepping through and trying to do manually using latest kpt

bash-5.1$ kpt pkg get https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages.git/asm@1.10.4-asm.6+config2 asm
Package "asm":
Fetching https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages@1.10.4-asm.6+config2
From https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages
 * tag               1.10.4-asm.6+config2 -> FETCH_HEAD
Error: Kptfile at "https:/github.com/GoogleCloudPlatform/anthos-service-mesh-packages/asm@1.10.4-asm.6+config2" has an old version ("v1alpha1") of the Kptfile schema.
Please update the package to the latest format by following https://kpt.dev/installation/migration. 
bash-5.1$ kpt version
1.0.0-beta.7

**Comment by edi-bice-by** [comment]
Not sure if it matters for kpt but Darwin was set to arm64 in the following code 

`download_kpt() {
  local OS

  case "$(uname)" in
    Linux ) OS="linux_amd64";;
    Darwin) OS="darwin_amd64";;
    *     ) fatal "$(uname) is not a supported OS.";;
  esac

  local KPT_TGZ
  KPT_TGZ="https://github.com/GoogleContainerTools/kpt/releases/download/v0.39.3/kpt_${OS}-0.39.3.tar.gz"

  info "Downloading kpt.."
  curl -L "${KPT_TGZ}" | tar xz
  AKPT="$(apath -f kpt)"
}`

**Comment by edi-bice-by** [comment]
Going through manually it was picking up the 1.0.0 kpt I installed as described in the docs. Seems for this particular step the 0.39 is needed so I changed to find it via relative path and used install_asm manually which worked

retry 3 ./kpt pkg get --auto-set=false "${KPT_URL}" asm

**Comment by bwassie** [comment]
Are you running this on MacOS by any chance? I had the same issue when using MacOS and I switched to linux and it worked fine.

**Comment by edi-bice-by** [comment]
Yes, I am though I upgraded to bash 5.1 and have been managing so far except for a few things here and there.
